plugins/heroku/src/buildHerokuReviewApp.ts

```diff
@@ -0,0 +1,42 @@
+import heroku from './herokuClient'
+import type { HerokuApiResGetReview, HerokuApiResPost } from 'heroku-client'
+import { Logger } from 'winston'
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { repeatedCheckForSuccessStatus } from './repeatedCheckForSuccessStatus'
+import { getRepoDetails } from './githubApi'
+
+async function buildHerokuReviewApp(
+	logger: Logger,
+	pipelineId: string
+	): Promise<string> {
+
+  const { branch, repo, source_blob } = await getRepoDetails(logger)
+
+  logger.info(`creating review app for ${branch} branch on ${repo}...`)
+  const reviewAppBuild: HerokuApiResPost = await heroku.post(`/review-apps`, {
+    body: {
+      branch: branch,
+      pipeline: pipelineId,
+      source_blob
+    }
+  })
+
+  logger.info('reviewApp in buildHerokuReviewApp', reviewAppBuild)
+
+  logger.info(`checking review app for success status...`)
+
+  const successStatus = await repeatedCheckForSuccessStatus(logger, reviewAppBuild.id)
+
+  const reviewApp: HerokuApiResGetReview = await heroku.get(`/review-apps/${reviewAppBuild.id}`)
```

Comment:
Hmm so none of the api calls in that function have a catch mechanism for their errors, however the function itself would then not resolve and therefore `review.ts` would throw. But maybe I should have a the whole function wrapped in a try/catch so that it would throw and carry the meaningful error message back to `review.ts`?